### PR TITLE
Add Excel97 Datatype

### PR DESCRIPTION
**Pull request by jmchilton**
Extension of #4200 that fixes Mac OS X compatibility and addressed Dan's concern about spaces in filenames. Merging this will merge #4200 as well.

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1120,6 +1120,39 @@ def sniff( self, filename ):
...
+        else:
+            return False
+
+    def get_mime( self ):
+        """Returns the mime type of the datatype"""
+        return 'application/vnd.ms-excel'
+
+    def set_peek( self, dataset, is_multi_byte=False ):
+        if not dataset.dataset.purged:
+            dataset.peek = "Microsoft Excel 2003 file"
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: s/2003/97/ ? Also below.

Comment (jmchilton) [reply lib/galaxy/datatypes/binary.py]: I can't tell if 2003 or 97 is better - maybe I'll just go with "Microsoft Excel XLS file" - https://msdn.microsoft.com/en-us/library/office/gg615597(v=office.14).aspx.

**Comment by bgruening** [comment]
This might also help: https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Fformat_3468
Can we also add the EDAM annotation of this datatype?

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/datatypes/binary.py]: > maybe I'll just go with "Microsoft Excel XLS file"

Sounds good.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**